render.go

```diff
@@ -0,0 +1,128 @@
+package render
+
+import (
+	"image/color"
+	"image/draw"
+	"math"
+
+	"github.com/benoitkugler/textlayout/fonts"
+	"github.com/go-text/typesetting/shaping"
+	"github.com/srwiley/rasterx"
+	"golang.org/x/image/math/fixed"
+)
+
+// Renderer defines a type that can render strings to a bitmap canvas.
+// The size and look of output depends on the various fields in this struct.
+// Developers should provide suitable output images for their draw requests.
+// This type is not thread safe so instances should be used from only 1 goroutine.
+type Renderer struct {
+	// FontSize defines the point size of output text, commonly between 10 and 14 for regular text
+	FontSize float32
+	// PixScale is used to indicate the pixel density of your output target.
+	// For example on a hi-DPI (or "retina") display this may be 2.0.
+	// Default value is 1.0, meaning
```

Comment:
I think maybe you meant to write more here?

Reply:
Oops, must have got side-tracked. Fixed